test/RelayPricer.test.ts

```diff
@@ -0,0 +1,86 @@
+import chai, { assert, expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { StubbedInstance, stubInterface } from 'ts-sinon';
+
+import { RelayPricer } from '../src';
+import { SourceApi } from '../src/types/RelayPricer';
+
+chai.use(chaiAsPromised);
+
+describe('RelayPricer', () => {
+    let fakeSourceApi: StubbedInstance<SourceApi>;
+    let pricer: RelayPricer;
+    const fakePrice1 = 23345.834630269488;
+    const fakePrice2 = 0.0775886924643371;
+    const fakePriceConversion1 = 300892.2291221775;
+    const fakePriceConversion2 = 0.0000033234490731694805;
+    const sourceCurrency = 'RBTC';
+    const targetCurrency = 'RIF';
+    const intermediateCurrency = 'USD';
+
+    describe('getPrice', async () => {
+        beforeEach(() => {
+            fakeSourceApi = stubInterface<SourceApi>();
+            pricer = new RelayPricer(fakeSourceApi);
+        });
+
+        it('should fail if receive wrong SourceApi', async () => {
+            expect(() => new RelayPricer('test')).to.throw(
+                'Source api not provided'
+            );
+        });
+
+        it('should return source currency conversion price', async () => {
+            fakeSourceApi.query
+                .withArgs(sourceCurrency, intermediateCurrency)
+                .returns(Promise.resolve(fakePrice1));
+            fakeSourceApi.query
+                .withArgs(targetCurrency, intermediateCurrency)
+                .returns(Promise.resolve(fakePrice2));
```

Comment:
it looks like this code is duplicated on line 45 as well. We may think about creating a function for doing this.

Reply:
Let me create a function

Reply:
updateds